internal/rpc/namenode.go

```diff
@@ -223,6 +223,24 @@ func (c *NamenodeConnection) Execute(method string, req proto.Message, resp prot
 	return nil
 }
 
+// GetEncryptionKeys will use the `getDataEncryptionKey` operation on the
+// namenode in order to fetch the current data encryption keys
+func (c *NamenodeConnection) GetEncryptionKeys() *EncryptionKey {
```

Comment:
I'm not sure about exporting this. Does it have utility to users?

Reply:
right now, i kept it exported because it's `Client` which is using it to pass to the `DatanodeSaslDialer` as the `KeyFunc` so that the dialer doesn't need to have a reference to the namenode connection directly, just a function which gives it the key. So since it's being used in the `client.go` file, it needs to be exported.

Reply:
> hm, maybe just pass the two options into NamenodeConnection and then put the wrapDatanodeDialer func there?

but then i'd have to export wrapDatanodeDialer since NamenodeConnection is in `internal/rpc` and this needs to be called from `file_reader` / `file_writer`. Either way something would need to get exposed since the block reader / writer don't have a reference to the namenode, which they shouldn't need anyways.

Reply:
I figured it made more sense to export this utility function which is just a function call on the Namenode than it would have been to export a wrapping function for the datanode dialer.